crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -865,6 +881,108 @@ where
         };
         build_transaction_receipt_with_block_receipts(tx, meta, receipt, &all_receipts)
     }
+
+    /// Helper function for `eth_getTransactionReceipt` (optimism)
+    ///
+    /// Returns the receipt
+    #[cfg(feature = "optimism")]
+    pub(crate) async fn build_transaction_receipt(
+        &self,
+        tx: TransactionSigned,
+        meta: TransactionMeta,
+        receipt: Receipt,
+    ) -> EthResult<TransactionReceipt> {
+        // get all receipts for the block
+        let all_receipts = match self.cache().get_receipts(meta.block_hash).await? {
+            Some(recpts) => recpts,
+            None => return Err(EthApiError::UnknownBlockNumber),
+        };
+        let block = self
+            .provider()
+            .block_by_number(meta.block_number)?
+            .ok_or(EthApiError::UnknownBlockNumber)?;
+
+        let l1_block_info = reth_revm::optimism::extract_l1_info(&block).ok().map(Rc::new);
+        let optimism_tx_meta = self.build_op_tx_meta(&tx, l1_block_info, block.timestamp)?;
+
+        build_transaction_receipt_with_block_receipts(
+            tx,
+            meta,
+            receipt,
+            &all_receipts,
+            optimism_tx_meta,
+        )
+    }
+
+    #[cfg(feature = "optimism")]
+    pub(crate) fn build_op_tx_meta(
```

Comment:
needs docs

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/155